core/trino-main/src/test/java/io/trino/operator/output/TestPositionsAppender.java

```diff
@@ -100,6 +102,80 @@ public void testNullRle(Type type)
+                input(notNullBlock(type, 2), 0, 1),
+                input(rleBlock(type, 3), 0, 1));
+        testAppend(type, inputs);
+
+        List<BlockView> dictionaryInputs = ImmutableList.of(
+                input(dictionaryBlock(type, 2, 4, 0.5F), 0, 1),
+                input(rleBlock(type, 3), 0, 1));
+        testAppend(type, dictionaryInputs);
+    }
+
+    @Test(dataProvider = "differentValues")
+    public void testMultipleRleBlocksWithDifferentValues(Type type, Block value1, Block value2)
+    {
+        List<BlockView> input = ImmutableList.of(
+                input(rleBlock(value1, 3), 0, 1),
+                input(rleBlock(value2, 3), 0, 1));
+        testAppend(type, input);
+    }
+
+    @DataProvider(name = "differentValues")
+    public static Object[][] differentValues()
+    {
+        return new Object[][]
+                {
+                        {BIGINT, createLongsBlock(0), createLongsBlock(1)},
+                        {BOOLEAN, createBooleansBlock(true), createBooleansBlock(false)},
+                        {INTEGER, createIntsBlock(0), createIntsBlock(1)},
+                        {createCharType(10), createStringsBlock("0"), createStringsBlock("1")},
+                        {createUnboundedVarcharType(), createStringsBlock("0"), createStringsBlock("1")},
+                        {DOUBLE, createDoublesBlock(0D), createDoublesBlock(1D)},
+                        {SMALLINT, createSmallintsBlock(0), createSmallintsBlock(1)},
+                        {TINYINT, createTinyintsBlock(0), createTinyintsBlock(1)},
+                        {VARBINARY, createSlicesBlock(Slices.wrappedLongArray(0)), createSlicesBlock(Slices.wrappedLongArray(1))},
+                        {createDecimalType(Decimals.MAX_SHORT_PRECISION + 1), createLongDecimalsBlock("0"), createLongDecimalsBlock("1")},
+                        {new ArrayType(BIGINT), createArrayBigintBlock(ImmutableList.of(ImmutableList.of(0L))), createArrayBigintBlock(ImmutableList.of(ImmutableList.of(1L)))},
+                        {
+                                createTimestampType(9),
+                                createLongTimestampBlock(createTimestampType(9), new LongTimestamp(0, 0)),
+                                createLongTimestampBlock(createTimestampType(9), new LongTimestamp(1, 0))}
+                };
+    }
+
+    @Test(dataProvider = "types")
+    public void testMultipleRleWithTheSameValueProduceRle(Type type)
+    {
+        PositionsAppender positionsAppender = POSITIONS_APPENDER_FACTORY.create(type, 10, DEFAULT_MAX_PAGE_SIZE_IN_BYTES);
+
+        Block value = notNullBlock(type, 1);
+        positionsAppender.append(allPositions(3), rleBlock(value, 3));
+        positionsAppender.append(allPositions(2), rleBlock(value, 2));
+
+        Block actual = positionsAppender.build();
+        assertEquals(actual.getPositionCount(), 5);
+        assertInstanceOf(actual, RunLengthEncodedBlock.class, actual.toString());
```

Comment:
you don't need to put `actual.toString()`. Here and in other `assertInstanceOf` method invocations